src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -1230,47 +1260,47 @@ void func_8089D318(EnDinofos* this, PlayState* play) {
             subCamEye.y = this->actor.focus.pos.y;
             subCamEye.z = (Math_CosS(this->actor.shape.rot.y) * 150.0f) + this->actor.focus.pos.z;
             Play_SetCameraAtEye(play, this->subCamId, &this->actor.focus.pos, &subCamEye);
-            func_8089CFAC(this);
+            EnDinofos_SetupDie(this);
         } else {
-            func_8089B100(this, play);
+            EnDinofos_SetupIntroCutsceneBeforeFall(this, play);
         }
     } else {
         CutsceneManager_Queue(this->actor.csId);
     }
 }
 
-void func_8089D42C(EnDinofos* this, PlayState* play) {
-    s16 temp_v0_2;
+void EnDinofos_RotateHead(EnDinofos* this, PlayState* play) {
+    s16 headRotYDelta;
 
-    if ((this->actionFunc == func_8089B834) && (this->unk_290 != 0)) {
-        Math_ScaledStepToS(&this->unk_28E, Math_SinS(this->unk_290 * 1400) * 0x2C00, 0x300);
-    } else if (this->actionFunc == func_8089CA74) {
-        Math_ScaledStepToS(&this->unk_28E, Math_CosF(M_PIf) * 0x2C00, 0x2C00 / 20);
-    } else if (this->actionFunc == func_8089CBEC) {
-        this->unk_28E = Math_CosF(this->unk_290 * (M_PIf / 20)) * 0x2C00;
+    if ((this->actionFunc == EnDinofos_Idle) && (this->headTimer != 0)) {
+        Math_ScaledStepToS(&this->headRotY, Math_SinS(this->headTimer * 1400) * 0x2C00, 0x300);
+    } else if (this->actionFunc == EnDinofos_StartBreatheFire) {
+        Math_ScaledStepToS(&this->headRotY, Math_CosF(M_PIf) * 0x2C00, 0x2C00 / 20);
+    } else if (this->actionFunc == EnDinofos_BreatheFire) {
+        this->headRotY = Math_CosF(this->headTimer * (M_PIf / 20)) * 0x2C00;
     } else if (!Play_InCsMode(play)) {
-        temp_v0_2 = this->unk_28E + this->actor.shape.rot.y;
-        temp_v0_2 = BINANG_SUB(this->actor.yawTowardsPlayer, temp_v0_2);
-        temp_v0_2 = CLAMP(temp_v0_2, -0x300, 0x300);
-        this->unk_28E += temp_v0_2;
-        this->unk_28E = CLAMP(this->unk_28E, -0x2C00, 0x2C00);
+        headRotYDelta = this->headRotY + this->actor.shape.rot.y;
```

Comment:
```suggestion
        headRotYDiff = this->headRotY + this->actor.shape.rot.y;
```
I believe diff is much more common through out the codebase